### ISSUE TITLE
Extending OpenSearch Build pipeline to Include Plugins Perf Tests

### ISSUE THREAD

**Issue by navneet1v** [opensearch-project/opensearch-build#2988]
### Is your feature request related to a problem? Please describe

We work on 3 plugins of OpenSearch. During every release of OpenSearch the release owner needs to run the perf tests for those plugins manually and only after the release candidate is created. This creates an extra effort for the release owner and is prone to bugs that cannot be fixed once the code freeze has happened.

### Describe the solution you'd like

The solution I was thinking is to extend the OS Jenkins pipeline used by OpenSearch to run the plugin level Perf Tests. There can be many ways in which the pipeline infrastructure can be extended :
1. Adding more stages in the same pipeline where each stage can belong to plugin owner
2. Plugin teams can build their own pipeline and they can get triggers from main OpenSearch build Pipeline.

### Describe alternatives you've considered

The only alternative that I can think of was to automate the setup for our plugins and build things from Scratch.

### Additional context

The perf tests which our plugin runs are build using both OpenSearch Benchmarks and without OpenSearch Benchmarks. In this feature request we want to first explore options for tests written using OSB but not limited to OSB only.

**Comment by rishabh6788** [comment]
As per my discussion with @gaiksaya and @peterzhuamazon the k-NN team should be able to add plugin specific performance tests, feel free to contribute like [CCR](https://github.com/opensearch-project/opensearch-build/blob/main/jenkins/cross-cluster-replication/perf-test.jenkinsfile)   
Regarding the ownership and maintenance of the job, would like @bbarani to comment on it. 

In the meantime it would be helpful if you could provide more details into your special requirements regarding the type of load you want to run and do you want it to be integrated with mensor tool?
@navneet1v

**Comment by navneet1v** [comment]
Hi @rishabh6788 Thanks a lot for providing the update. We will look into this Code link provided by you.

Requirements that I can think of as of now:
1. For K-NN plugin we have atleast 4 opensearch benchmark workloads that we will be running. Code link: https://github.com/opensearch-project/k-NN/tree/main/benchmarks/osb
2. For K-NN workloads as of today we run on 3 master and 3 data nodes on r5.4xlarge machines.

I am not sure if there is any other requirements we have as of now for k-NN. During the integration if something comes up we can discuss.

From mensor tool integration, yes we want to integrate with mensor tool.


Apart from this, can you provide the details around how to test the jobs that we will be creating?

**Comment by gaiksaya** [comment]
[Triage] Adding @bbarani to add more input about the ownership and infrastructure support for this.
Thanks!

**Comment by bbarani** [comment]
@rishabh6788 Can you add your inputs?

**Comment by rishabh6788** [comment]
@navneet1v We recently released [opensearch.org/benchmark ](https://opensearch.org/benchmarks) to run nightly benchmark tests on the latest build of 2.8, 2.9 and 3.0. 
To run plugin specific perf test we would need a workload created just to target that particular plugin which we currently do not have. 
Can you please let me know if you already have a workload created that tests a particular plugin and if that can be added to https://github.com/opensearch-project/opensearch-benchmark-workloads repo so that the same can be used to run nightly benchmarks?

Action item on us would be  tomake sure before we run the plugin specific benchmark, that plugin is present in the bundle.

**Comment by navneet1v** [comment]
@rishabh6788 Yes we have a workload, but currently it is present in the K-NN repo. 
Workload: https://github.com/opensearch-project/k-NN/tree/main/benchmarks/osb

**Comment by rishabh6788** [comment]
@navneet1v Is there any plan to add that workload to https://github.com/opensearch-project/opensearch-benchmark-workloads repo?
Once the workload is added it can be added to nightly benchmark runs.

**Comment by navneet1v** [comment]
@vamshin can we scope this out. So that K-NN opensearch benchmarks can run nightly.

**Comment by rishabh6788** [comment]
@navneet1v Can you please let us know what is the status on migrating your workload to https://github.com/opensearch-project/opensearch-benchmark-workloads repo? Please let me know if you need any help from benchmark team to migrate the workload. 
Once the workload has been added to the workload repo we can enable nightly runs.

**Comment by navneet1v** [comment]
@rishabh6788 We are working on other high priority tasks for now. I have already communicated this to @vamshin. He needs to plan for this task.
